src/Action.cpp

```diff
@@ -118,71 +117,137 @@ std::string DuplicateUser::toString() const {
 }
 
 
-void Exit::act(Session& sess) {
+void PrintContentList::act(Session& sess) {
+    const std::vector<Watchable*>& content = sess.getContent();
+    int i = 1; //Because human lists start at 1
+    for(auto w: content)
+    {
+        std::cout << std::to_string(i) << w->toString() << std::endl;
+        i++;
+    }
+}
+
+std::string PrintContentList::toString() const {
+    return "PrintContentList " + getStatusString();
+}
+
+void PrintWatchHistory::act(Session& sess) {
+    const std::vector<Watchable*> history = sess.getActiveUser().get_history();
+    int i = 1; //Because human lists start at 1
+    for(auto w: history)
+    {
+        std::cout << std::to_string(i) << w->toString() << std::endl;
+        i++;
+    }
     complete();
 }
 
-//TODO check if this is the right printing format
-std::string Exit::toString() const {
-    return "Exit "+getStatusString();
+std::string PrintWatchHistory::toString() const {
+    return "PrintWatchHistory " + getStatusString();
 }
 
-PrintActionsLog::PrintActionsLog(const std::vector<BaseAction*>& actionsLog): actionsLog(actionsLog) {}
+Watch::Watch(Watchable& toWatch): BaseAction(),
+                                 _toWatch(toWatch),
+                                  _nextWatchableId(-1) {}
+
+void Watch::act(Session& sess) {
+    User& activeUser = sess.getActiveUser();
+    std::cout << "Watching " << _toWatch.toString();
+    activeUser.addToHistory(_toWatch);
+    Watchable* nextWatchable = _toWatch.getNextWatchable(sess);
```

Comment:
probably you can put the complete here between the two lines. I don't know if they'd like it so you can just call it twice